consensus/types/src/blob_sidecar.rs

```diff
@@ -1,17 +1,109 @@
-use crate::test_utils::TestRandom;
-use crate::{Blob, ChainSpec, Domain, EthSpec, Fork, Hash256, SignedBlobSidecar, SignedRoot, Slot};
-use bls::SecretKey;
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
 use derivative::Derivative;
 use kzg::{Kzg, KzgCommitment, KzgPreset, KzgProof};
 use rand::Rng;
+use serde::de::DeserializeOwned;
 use serde_derive::{Deserialize, Serialize};
-use ssz::Encode;
+use ssz::{Decode, Encode};
 use ssz_derive::{Decode, Encode};
 use ssz_types::{FixedVector, VariableList};
-use std::sync::Arc;
-use test_random_derive::TestRandom;
+use tree_hash::TreeHash;
 use tree_hash_derive::TreeHash;
 
+use bls::SecretKey;
+use test_random_derive::TestRandom;
+
+use crate::beacon_block_body::KzgCommitments;
+use crate::test_utils::TestRandom;
+use crate::{
+    AbstractExecPayload, BeaconBlock, Blob, ChainSpec, Domain, EthSpec, Fork, Hash256,
+    SignedBlobSidecar, SignedRoot, Slot,
+};
+
+pub trait Sidecar<E: EthSpec>:
+    serde::Serialize
+    + Clone
+    + DeserializeOwned
+    + Encode
+    + Decode
+    + Hash
+    + TreeHash
+    + TestRandom
+    + Debug
+    + SignedRoot
+    + Sync
+    + Send
+    + for<'a> arbitrary::Arbitrary<'a>
+{
+    type RawBlobs: RawBlobs<E>;
```

Comment:
I know this way my name, but I think it's confusing - it makes me think of raw bytes but it's more like "list of blobs that aren't in sidecars", which I don't currently have a good name for... what about going for `Blobs` here and renaming these type aliases:

```
pub type BlobsList<T> = VariableList<Blob<T>, <T as EthSpec>::MaxBlobsPerBlock>;
pub type BlobRootsList<T> = VariableList<Hash256, <T as EthSpec>::MaxBlobsPerBlock>;
```

Reply:
Renaming the trait to `BlobItems` and the type aliases to the above suggestions.  Please see commit d3befb05fa1133fe92d7723f735f8bdb19c35f69.